internal/serverinstall/k8s.go

```diff
@@ -379,139 +337,127 @@ func (i *K8sInstaller) Upgrade(
 		}
 	}
 
-	ui := opts.UI
 	log := opts.Log
+	ui := opts.UI
 
 	sg := ui.StepGroup()
 	defer sg.Wait()
 
-	s := sg.Add("Inspecting Kubernetes cluster...")
-	defer s.Abort()
-
-	clientset, err := i.newClient()
+	s := sg.Add("Getting Helm configs...")
+	defer func() { s.Abort() }()
+	settings, err := helminstallutil.SettingsInit()
 	if err != nil {
-		ui.Output(err.Error(), terminal.WithErrorStyle())
 		return nil, err
 	}
+	s.Update("Helm settings retrieved")
+	s.Status(terminal.StatusOK)
+	s.Done()
 
-	// Do some probing to see if this is OpenShift. If so, we'll switch the config for the user.
-	// Setting the OpenShift flag will short circuit this.
-	if !i.config.openshift {
-		s.Update("Gathering information about the Kubernetes cluster...")
-
-		namespaceClient := clientset.CoreV1().Namespaces()
-		_, err := namespaceClient.Get(context.TODO(), "openshift", metav1.GetOptions{})
-		isOpenShift := err == nil
-
-		// Default namespace in OpenShift acts like a regular K8s namespace, so we don't want
-		// to remove fsGroup in this case.
-		if isOpenShift && i.config.namespace != "default" {
-			s.Update("OpenShift detected. Switching configuration...")
-			i.config.openshift = true
-		}
```

Comment:
@briancain the Openshift flag/auto-detection is no longer useful since we're installing to k8s with the Helm chart. Before this branch, it appears that not detecting Openshift influenced the pod security context for the k8s deployment that was created:

```
	securityContext := &apiv1.PodSecurityContext{}
	if !c.openshift {
		securityContext.FSGroup = int64Ptr(1000)
	}
```

I also double checked in the Helm chart and it looks like we don't have any values to set related to Openshift or pod security context, so I think this is safely removed:

https://github.com/hashicorp/waypoint-helm/blob/main/values.yaml